### ISSUE TITLE
508-defect-4 [AXE CORE]: ID in ARIA and labels must be unique

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/va.gov-team#34252]
[508-defect-4 ✔️ Trivial](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-4)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _First name only. Delete if not applicable._

## Axe Violation

**[ID attribute values must be unique - Minor](https://dequeuniversity.com/rules/axe/3.5/duplicate-id?application=coconut)**
The ID attribute uniquely identifies elements on a page. It does not make sense to duplicate an ID.

Duplicate ID's can break the accessibility of labels for forms, table header cells, etc., by the second instance being skipped by screen readers, or by client-side scripts. They are common markup validation errors that can eliminate possible sources of accessibility problems, when not breaking accessibility.

Those experienced with client-side scripting know that when you re-use an id, typically the only one that gets acted upon by the scripting is the first instance of the use of that ID. Similarly, assistive technologies may, when referencing an id, only reference the first one accurately.

## Acceptance Criteria

- [x] Axe no longer flags a violation for the error message

## Steps to Recreate

1. Go to https://staging.va.gov in browser
1. Log in as any user
1. Open the developer tools panel (Press <kbd>F12</kbd>)
1. Toggle the device toolbar (phone/tablet icon in the upper-left corner)
1. Set the width dimension to less than 768 to collapse the menu
1. Run axe (see screenshot) - the `#account-menu` id occurs in 2 places

### Duplicate ID

<img width="691" alt="Screen Shot 2021-12-16 at 10 58 10 AM" src="https://user-images.githubusercontent.com/136959/146415631-a8a01ebc-f9fd-40f3-9cdb-8968ac20bb41.png">

### Axe error

<img width="1577" alt="Screen Shot 2021-12-16 at 9 50 25 AM" src="https://user-images.githubusercontent.com/136959/146415616-95a9c707-8d96-4031-8c7f-ff76af6a49c1.png">

**Comment by brianalloyd** [comment]
@tmac9494 ticket to pick up this sprint. Let me know what questions exist.

**Comment by kelsonic** [comment]
This is not an easy one... So the reason we are getting a duplicate ID is because the legacy header is still on the DOM when header v2 is rendered, the legacy header is just display: none;. Hey @joshkimux would adding `aria-hidden="true"` resolve this issue when the legacy header is hidden for header v2?

![image](https://user-images.githubusercontent.com/12773166/146418491-76017926-9234-4cf9-9ac7-5f5dc96a978c.png)

**Comment by joshkimux** [comment]
This was a good catch from Robin, and it's important to flag these! I did a little more digging after recommending Robin to create a `defect-1` ticket on this.

Technically with `vads-u-display--none` (which I'm guessing is `display:none`), the legacy header _should_ be hidden from the accessibility API which means there shouldn't be an issue with duplicate ids. That being said, the issue will be flagged by axe core regardless. 

For starters I'll bump down my initial rating from a `defect-1` to a `defect-4`, as this _shouldn't_ cause any issues related to duplicate IDs when it is hidden. @kelsonic , to remove the issue from emerging again, would we be able to give the legacy header a unique id? e.g. `id="legacy-header-1"`?

**Comment by kelsonic** [comment]
Both the legacy header and the new header use the same component for the auth menu, so the auth menu doesn't know about its id that's a duplicate 😬

**Comment by joshkimux** [comment]
@kelsonic ooph that's tricky 😅 
If this doesn't cause any actual a11y issues (since it should be hidden from the a11y tree), I'm inclined to closing it now or leaving it open as a `defect-4` until the legacy header is officially retired so folks know why the issue is being flagged in axe. 

That being said, I'm still not too sure how to proceed here then-- @noahgelman would you be able to help us?

**Comment by noahgelman** [comment]
@joshkimux Hmm, unique situation. Why is the legacy header hidden instead of removed? Is it being dynamically hidden/shown? If so, is it possible to to remove/add the idea at the same time? (weird solution I know)
